src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/MachObjectWriter.cs

```diff
@@ -347,6 +347,10 @@ private protected override void CreateSection(ObjectNodeSection section, string
             {
                 ".dotnet_eh_table" => S_ATTR_DEBUG,
                 ".eh_frame" => S_ATTR_LIVE_SUPPORT | S_ATTR_STRIP_STATIC_SYMS | S_ATTR_NO_TOC,
+                // workaround problem with dead_strip
+                // see https://github.com/dotnet/runtime/issues/96663
+                "hydrated" => S_ATTR_NO_DEAD_STRIP,
```

Comment:
Hydration target section is essentially BSS, or uninitalized data. The failures in the CI are from the conflicting definition of uninitialized data (ie. no actual data are in the file, just size) and `S_ATTR_NO_DEAD_STRIP`.

The linker cannot strip something that's not there in the first place, so this is invalid combination.

Reply:
Umm... In my mac, without `S_ATTR_NO_DEAD_STRIP` on the `hydrated` section, another memory error that is looks hydrated data is broken.

I found several symbols in the `hydrated` section are removed and size of `hydrated` section is changed so I assumed the stripping symbols and spaces in the `hydrated` will cause problem with __dehydrated_data data.

Reply:
I'm not entirely familiar with the format of dehydrated data. Out of curiosity, did you try running the build with `IlcDehydrate=false`?

Reply:
> did you try running the build with IlcDehydrate=false?

I just tried with `IlcDehydrate=false` and it works.